packages/api/src/incentives/component/componentWhitelistRouter.ts

```diff
@@ -0,0 +1,111 @@
+import { createTRPCRouter, publicProcedure } from "../trpc";
+import { TRPCError } from "@trpc/server";
+import { z } from "zod";
+import { Exit } from "effect";
+
+const csvUploadSchema = z.object({
+  csvData: z.string(),
+});
+
+export const componentWhitelistRouter = createTRPCRouter({
+  uploadCsv: publicProcedure
+    .input(csvUploadSchema)
+    .mutation(async ({ ctx, input }) => {
+      try {
```

Comment:
- move logic to effect and expose through dependency layer.
- add unit test to make sure that the csv parsing logic works.
- don't use try/catch, use Exit